src/AzSK.ADO/Framework/Core/STMapping/AzSKADOServiceMapping.ps1

```diff
@@ -326,13 +352,56 @@ class AzSKADOServiceMapping: CommandBase
                         if(($varGrps | Measure-Object).Count -gt 0)
                         {
                         $varGrps | ForEach-Object{
-                            $varGrpURL = ("https://{0}.visualstudio.com/{1}/_apis/distributedtask/variablegroups/{2}?api-version=6.1-preview.2") -f $this.OrgName, $this.projectId, $_;
-                            $varGrpObj = [WebRequestHelper]::InvokeGetWebRequest($varGrpURL);
+                            $varGrpURL = ("https://{0}.visualstudio.com/{1}/_apis/distributedtask/variablegroups/{2}?api-version=6.1-preview.2") -f $this.OrgName, $this.projectId, $_;                            
+                            $varGrpObj = [WebRequestHelper]::InvokeGetWebRequest($varGrpURL);                              
+                                if ($varGrpObj.Type -eq 'AzureKeyVault')
```

Comment:
First we should take mapping from existing logic as it is taking from Release ST mapping(no api call there), if not found then only try to get